pkg/controllers/provisioning/launcher.go

```diff
@@ -111,6 +118,21 @@ func (l *Launcher) bind(ctx context.Context, node *v1.Node, pods []*v1.Pod) (err
 	return nil
 }
 
+func (l *Launcher) verifyResourceLimits(ctx context.Context, provisioner *v1alpha5.Provisioner) error {
+	provisionerLatest := &v1alpha5.Provisioner{}
+	if err := l.KubeClient.Get(ctx, types.NamespacedName{Name: provisioner.Name, Namespace: provisioner.Namespace}, provisionerLatest); err != nil {
+		return err
+	}
+	for resourceName, usage := range provisionerLatest.Status.Resources {
+		if limit, ok := provisionerLatest.Spec.Limits.Resources[resourceName]; ok {
```

Comment:
I thought of using provisioner.Spec.Limits here but figured that if we hot update the limits it should be fine

Reply:
I'm a little nervous of this, since it violates assumptions about which policy is applied. I'd just used provisioner.Spec.Limits to reduce the complexity surface.